File: frontend/src/api/foodComment.ts
/**
 * 美食评论 API
 */
import request from './index';

export interface FoodComment {
  id: number;
  foodId: number;
  userId: number;
  userName?: string;
  rating: number;
  content: string;
  likes: number;
  isLiked?: boolean;
  parentId?: number;
  replyToUserId?: number;
  replyToUser?: string;
  replies?: FoodComment[];
  createdAt: string;
}

export interface AddCommentReq {
  userId: number;
  userName?: string;
  rating: number;
  content: string;
}

export interface ReplyCommentReq {
  userId: number;
  userName?: string;
  content: string;
}

/**
 * 获取美食评论列表
 */
export function getFoodComments(foodId: number) {
  return request.get<FoodComment[]>(`/foods/${foodId}/comments`);
}

/**
 * 添加美食评论
 */
export function addFoodComment(foodId: number, data: AddCommentReq) {
  return request.post(`/foods/${foodId}/comments`, data);
}

/**
 * 点赞评论
 */
export function likeFoodComment(commentId: number) {
  return request.post(`/food-comments/${commentId}/like`);
}

/**
 * 回复评论
 */
export function replyFoodComment(commentId: number, data: ReplyCommentReq) {
  return request.post(`/food-comments/${commentId}/reply`, data);
}

/**
 * 获取评论的回复列表
 */
export function getFoodCommentReplies(commentId: number) {
  return request.get<FoodComment[]>(`/food-comments/${commentId}/replies`);
}

/**
 * 删除评论
 */
export function deleteFoodComment(commentId: number) {
  return request.delete(`/food-comments/${commentId}`);
}

File: backend/src/main/java/com/learningassistant/backend/modules/food/model/FoodComment.java
package com.learningassistant.backend.modules.food.model;

import jakarta.persistence.*;
import lombok.Data;
import java.time.LocalDateTime;
import java.util.List;

/**
 * 美食评论实体
 */
@Entity
@Table(name = "food_comments")
@Data
public class FoodComment {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "food_id", nullable = false)
    private Long foodId;

    @Column(name = "user_id", nullable = false)
    private Long userId;

    @Column(name = "user_name")
    private String userName;

    @Column(nullable = false)
    private Integer rating;

    @Column(columnDefinition = "TEXT")
    private String content;

    @Column(columnDefinition = "INTEGER DEFAULT 0")
    private Integer likes = 0;
    
    // 父评论ID（用于回复功能）
    @Column(name = "parent_id")
    private Long parentId;
    
    // 回复的用户ID
    @Column(name = "reply_to_user_id")
    private Long replyToUserId;
    
    // 回复的用户名
    @Column(name = "reply_to_user")
    private String replyToUser;

    @Column(name = "created_at")
    private LocalDateTime createdAt = LocalDateTime.now();
    
    // 回复列表（非持久化字段，用于返回给前端）
    @Transient
    private List<FoodComment> replies;
}
